PERFORMANCE

```diff
@@ -37,6 +37,55 @@ To characterize Chapel performance, generally speaking...
   assignments are used to move chunks of data between locales.
 
 
+Experimental flags for improving performance
+--------------------------------------------
+
+The following compiler flags are intended to provide a preview of
+performance improvements that we anticipate being delivered by the
+compiler automatically as it matures.  Both are available for use "at
+your own risk" in that they are not guaranteed to maintain program
+correctness (details available after each flag).
+
+* chpl -sassertNoSlicing=true ...
+
+  Chapel's arrays are currently heavyweight compared to C/Fortran in
+  order to support Chapel's rich array semantics.  Specifically, in
+  the most general case, Chapel's support for striding, reindexing,
+  and rank-change of arrays can require a multiplication when indexing
+  the innermost dimension of an array; in contrast, C and Fortran do
+  not require this multiplication.  For memory-bound programs, this
+  multiplication is rarely noticeable, but for programs that are
+  well-tuned for the memory hierarchy, this extra multiplication can
+  have a significant impact.
+
+  Work is currently underway to automatically distinguish between
+  arrays that require this multiplication and those that don't in
+  order to remove the overhead in the (common) case when it is
+  unnecessary.  In the meantime, one can request that this
+  multiplication never be used within a program by throwing this flag.
+  The flag should be safe for any programs that do not reindex a
+  strided array or perform rank changes on an array that remove the
+  innermost dimension.
```

Comment:
For the sake of simplicity, I'm not going to bother and am hoping that the rephrasing of the original sentence to emphasize "indexing arrays" will cover us.